Tests/SentryTests/Swift/Core/Tools/LoadValidatorTests.swift

```diff
@@ -0,0 +1,197 @@
+        XCTAssertTrue(testOutput.loggedMessages.contains { $0.contains("⚠️ This can cause undefined behavior, crashes, or duplicate reporting.") })
+        XCTAssertTrue(testOutput.loggedMessages.contains { $0.contains("Ensure the SDK is linked only once, found `SentryDependencyContainerSwiftHelper` class in image path: \(imageName)") })
+        XCTAssertEqual(dispatchQueueWrapper.dispatchAsyncInvocations.count, 1)
+    }
+
+    func testValidateSDKPresenceIn_ContainsTargetClassNameInCurrentImage_SkipsValidation() {
+        // Arrange
+        let imageName = "/path/to/sentry/libSentry.dylib"
+        let loadValidatorAddress = LoadValidator.getCurrentFrameworkTextPointer()
+        let loadValidatorAddressValue = UInt(bitPattern: loadValidatorAddress)
+        let imageAddress = UInt64(loadValidatorAddressValue)
+        let imageSize: UInt64 = 0x1000
+        var getClassListCalled = false
+        testObjCRuntimeWrapper.beforeGetClassList = {
+            getClassListCalled = true
+        }
+        testObjCRuntimeWrapper.classesNames = { _ in
+            return ["SentryDependencyContainerSwiftHelper"]
+        }
+        
+        // Act
+        let validationResult = LoadValidator.checkForDuplicatedSDKInSync(imageName, imageAddress, imageSize,
+                                                                         objcRuntimeWrapper: testObjCRuntimeWrapper,
+                                                                       dispatchQueueWrapper: dispatchQueueWrapper)
+        
+        // Assert
+        XCTAssertFalse(validationResult, "Validation should skip for sentry framework")
+        XCTAssertTrue(getClassListCalled, "ObjectiveC Wrapper should not be called for an app binary")
+        XCTAssertFalse(testOutput.loggedMessages.contains { $0.contains("❌ Sentry SDK was loaded multiple times") })
+        XCTAssertEqual(dispatchQueueWrapper.dispatchAsyncInvocations.count, 1)
+    }
+}
+
+private extension LoadValidator {
+    /**
+     * This synchronous version is intended to be used in tests.
+     */
+    @discardableResult class func checkForDuplicatedSDKInSync(_ imageName: String,
+                                                              _ imageAddress: UInt64,
+                                                              _ imageSize: UInt64,
+                                                              objcRuntimeWrapper: SentryObjCRuntimeWrapper,
+                                                              dispatchQueueWrapper: SentryDispatchQueueWrapper) -> Bool {
+        var result = false
+        let semaphore = DispatchSemaphore(value: 0)
+        internalCheckForDuplicatedSDK(imageName, imageAddress, imageSize, objcRuntimeWrapper: objcRuntimeWrapper,
+                                      dispatchQueueWrapper: dispatchQueueWrapper) { duplicateFound in
+            result = duplicateFound
+            semaphore.signal()
+        }
+        semaphore.wait()
```

Comment:
`m`: We could replace this now with test expectations and use a timeout. Otherwise, we could block the tests indefinitely.